/***************************************************************************
 *  The FreeMedForms project is a set of free, open source medical         *
 *  applications.                                                          *
 *  (C) 2008-2015 by Eric MAEKER, MD (France) <eric.maeker@gmail.com>      *
 *  All rights reserved.                                                   *
 *                                                                         *
 *  This program is free software: you can redistribute it and/or modify   *
 *  it under the terms of the GNU General Public License as published by   *
 *  the Free Software Foundation, either version 3 of the License, or      *
 *  (at your option) any later version.                                    *
 *                                                                         *
 *  This program is distributed in the hope that it will be useful,        *
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of         *
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
 *  GNU General Public License for more details.                           *
 *                                                                         *
 *  You should have received a copy of the GNU General Public License      *
 *  along with this program (COPYING.FREEMEDFORMS file).                   *
 *  If not, see <http://www.gnu.org/licenses/>.                            *
 ***************************************************************************/
/***************************************************************************
 *   Main Developers:                                                     *
 *       Eric MAEKER, MD <eric.maeker@gmail.com>                           *
 *  Contributors:                                                          *
 *       NAME <MAIL@ADDRESS.COM>                                           *
 ***************************************************************************/
#ifndef UTILS_LOGINWIDGET_H
#define UTILS_LOGINWIDGET_H

#include <utils/global_exporter.h>
#include <QWidget>

/**
 * \file loginwidget.h
 * \author Eric Maeker
 * \version 0.8.4
 * \date 04 May 2013
*/

namespace Utils {

namespace Ui {
    class LoginWidget;
}

class UTILS_EXPORT LoginWidget : public QWidget
{
    Q_OBJECT
public:
    explicit LoginWidget(QWidget *parent = 0);
    ~LoginWidget();

    void setToggleViewIcon(const QString &fullAbsPath);
    void focusLogin();

    void toggleLoginEcho(bool visible);
    void togglePasswordEcho(bool visible);

    QString login() const;
    QString password() const;


private:
    void changeEvent(QEvent *e);

private:
    Ui::LoginWidget *ui;
};

} // namespace Utils

#endif // UTILS_LOGINWIDGET_H
